src/Hosting/Hosting/src/GenericHost/GenericWebHostBuilder.cs

```diff
@@ -390,13 +390,13 @@ private WebHostBuilderContext GetWebHostBuilderContext(HostBuilderContext contex
         if (!context.Properties.TryGetValue(typeof(WebHostBuilderContext), out var contextVal))
         {
             // Use _config as a fallback for WebHostOptions in case the chained source was removed from the hosting IConfigurationBuilder.
-            var options = new WebHostOptions(context.Configuration, fallbackConfiguration: _config);
+            var options = new WebHostOptions(context.Configuration, fallbackConfiguration: _config, environment: context.HostingEnvironment);
```

Comment:
How does this affect existing applications not using the WebApplicationBuilder?

Reply:
I just wanted to be 1000% sure that the `GenericWebHostBuilder` and the `HostBuilder`/`HostApplicationBuilder` could never be split-brained about environmental config (`ApplicationName`, `Environment` and `ContentRootPath`) since that's historically been a source of bugs.

You'll notice that we already do exactly this on line `399` where we already passed in `context.HostingEnvironment.ContentRootPath` directly instead of using `options.ContentRootPath` which we could easily have used instead.

Changing

```suggestion
            var options = new WebHostOptions(context.Configuration, fallbackConfiguration: _config, environment: null);
```

and changing `baseEnvironment: context.HostingEnvironment` to `baseEnvironment: null` below causes [this test](https://github.com/dotnet/aspnetcore/blob/49270eeadd9830c0f8f0ffda5fcd02754da518a7/src/DefaultBuilder/test/Microsoft.AspNetCore.Tests/WebApplicationTests.cs#L513) to fail.

```
  Error Message:
   Assert.Equal() Failure
          ↓ (pos 0)
Expected: WebApplicationTests_ENV
Actual:   Test
          ↑ (pos 0)
```


Technically only one of these lines need to change to fix the test, but I like being super explicit about the source of truth.